tests/test_upload.py

```diff
@@ -178,12 +180,7 @@ def none_upload(*args, **kwargs):
                "TWINE_CERT": "/foo/bar.crt"}
     with helpers.set_env(**testenv):
         cli.dispatch(["upload", "path/to/file"])
-    cli.dispatch(["upload", "path/to/file"])
```

Comment:
With much of the work separated out into the Settings object which is created and managed outside of the `upload.upload` function, our tests will error on this call since the Settings object otherwise attempts to retrieve the missing values.